### ISSUE TITLE
Add nofollow to Post-Meta Links Or Remove Links

### ISSUE THREAD

**Issue by hahntech** [Nikeo/customizr#1244]
I posted this on wordpress.org customizr forum. Then realized github get more traffic so I am repeating the post here:

This could be a feature request or a request for a custom solution.

I want to be able to display the post metas without search engines trying to follow the links that each meta tag includes. One method would be to add rel=”nofollow” to all the links associated with post metas. The other would be to display the post meta as text only, completely removing the associated link.

I would prefer the later solution over the former.

Why? Because I use a plugin that adds a Q&A forum to my site and I don’t want the post metas for all the forum content to be indexed or followed by search engines. Because most of them create 404 errors. Which means they are of no use for human visitors to the site.

Thanks for any tips you can provide!

**Comment by hahntech** [comment]
I would really appreciate some help with this one. Not sure where else to get assistance with this. I was able to locate the php file: core/front/models/content/post-metas/class-model-post_metas.php

And I was able to locate and modify the portion of the function that inserts the url link into the author-meta element. 

I made changes to the function named: czr_fn_get_meta_author()

Here is the original code snippet beginning at line 124:
`return apply_filters(
        'tc_author_meta',
        sprintf( '<span class="author vcard author_name"><a class="url fn n" href="%1$s" title="%2$s" rel="author">%3$s</a></span>' ,
            esc_url( get_author_posts_url( get_the_author_meta( 'ID', $author_id ) ) ),
            esc_attr( sprintf( __( 'View all posts by %s' , 'customizr' ), get_the_author_meta( 'display_name', $author_id ) ) ),
            get_the_author_meta( 'display_name', $author_id )
        )
    );//end filter`

And here is the modified version:
`return apply_filters(
        'tc_author_meta',
        sprintf( '<span class="author vcard author_name">%3$s</span>' ,
            esc_url( get_author_posts_url( get_the_author_meta( 'ID', $author_id ) ) ),
            esc_attr( sprintf( __( 'View all posts by %s' , 'customizr' ), get_the_author_meta( 'display_name', $author_id ) ) ),
            get_the_author_meta( 'display_name', $author_id )
        )
    );//end filter`

(edit: I have no idea why the formatting was not retained for the code I pasted above)

So I can figure out how to do the same for entry-meta and others. Now I just need to know how to make this work in my child theme so that it will not be erased by updates. 

Please and thank you.

**Comment by hahntech** [comment]
I have a potential solution to this issue. I was able to take a copy of the file I listed above 'class-model-post-metas.php' and place that in my child theme using the same directory structure. I then modified the file to check if the user level of the post 'author' is set to subscriber. If so, I strip out the href. Otherwise, normal behavior is performed. 

The reason for this is that when adding a plugin called CM Answers Pro, the users are able to login and register using social login (oAuth). I default all new registrations at subscriber level. So all my forum participants are held at subscriber level. Which is what created this issue in the first place. Every time a new post is added to the forum, the user gets tagged as the 'author' of the post. Even though they do not have Author level privileges. 

When a single post from the Q&A forum is displayed, Customizr sets the post metas, including links, for the 'author' name and the entry date. These links create 404's because they are constructed making the assumption each post in the Q&A forum is a blog post. 

The attached file contains the modifications that correctly remove the links from the post metas when the user is set to subscriber level. The modifications were made to the following two functions: czr_fn_get_meta_date() and czr_fn_get_meta_author().

[class-model-post-metas.txt](https://github.com/presscustomizr/customizr/files/1462480/class-model-post-metas.txt)
